neonvm/controllers/virtualmachine_qmp_queries.go

```diff
@@ -327,19 +410,19 @@ func QmpSyncMemoryToTarget(vm *vmv1.VirtualMachine, migration *vmv1.VirtualMachi
 			continue
 		}
 		// add memdev object
-		memdevId := strings.ReplaceAll(m.Data.Memdev, "/objects/", "")
-		cmd := []byte(fmt.Sprintf(`{"execute": "object-add", "arguments": {"id": "%s", "size": %d, "qom-type": "memory-backend-ram"}}`, memdevId, m.Data.Size))
-		_, err = target.Run(cmd)
+		memdevIdx, err := QmpMatchMemBackend(m.Data.Memdev)
+		if err != nil {
+			return err
+		}
+		err = QmpAddMemoryBackend(target, memdevIdx, m.Data.Size)
 		if err != nil {
 			return err
 		}
 		// now add pc-dimm device
-		cmd = []byte(fmt.Sprintf(`{"execute": "device_add", "arguments": {"id": "%s", "driver": "pc-dimm", "memdev": "%s"}}`, m.Data.Id, memdevId))
-		_, err = target.Run(cmd)
+		err = QmpAddMemoryDevice(target, memdevIdx)
 		if err != nil {
 			// device_add command failed... so try remove object that we just created
-			cmd = []byte(fmt.Sprintf(`{"execute": "object-del", "arguments": {"id": "%s"}}`, m.Data.Memdev))
-			target.Run(cmd)
+			_ = QmpDelMemoryBackend(target, memdevIdx)
```

Comment:
not a blocker: Do we need to also `QmpDelMemoryDevice` here?

I know the previous version doesn't, but is it possible for e.g. adding the memory device to "fail" due to timeout, but the operation to go through? I guess: what happens if we try to remove the backend while the device is still present?